service/history/ndc/history_replicator_test.go

```diff
@@ -380,3 +382,1110 @@ func TestApplyEvents(t *testing.T) {
 	// Intentionally panic result. Will test applyEvents function seperately
 	assert.Panics(t, func() { replicator.ApplyEvents(nil, nil) })
 }
+
+func Test_applyEvents_EventTypeWorkflowExecutionStarted(t *testing.T) {
+	workflowExecutionStartedType := types.EventTypeWorkflowExecutionStarted
+
+	tests := map[string]struct {
+		mockExecutionCacheAffordance  func(mockExecutionCache *execution.MockCache)
+		mockReplicationTaskAffordance func(mockReplicationTask *MockreplicationTask)
+		applyStartEventsFnAffordance  func() func(
+			ctx ctx.Context,
+			context execution.Context,
+			releaseFn execution.ReleaseFunc,
+			task replicationTask,
+			domainCache cache.DomainCache,
+			newMutableState newMutableStateFn,
+			newStateBuilder newStateBuilderFn,
+			transactionManager transactionManager,
+			logger log.Logger,
+			shard shard.Context,
+			clusterMetadata cluster.Metadata,
+		) error
+		expectError error
+	}{
+		"Case1: success case": {
+			mockExecutionCacheAffordance: func(mockExecutionCache *execution.MockCache) {
+				mockExecutionCache.EXPECT().GetOrCreateWorkflowExecution(gomock.Any(), gomock.Any(), gomock.Any()).
+					Return(nil, func(err error) {}, nil).Times(1)
+			},
+			mockReplicationTaskAffordance: func(mockReplicationTask *MockreplicationTask) {
+				mockReplicationTask.EXPECT().getDomainID().Return("test-domain-id").Times(1)
+				mockReplicationTask.EXPECT().getExecution().Return(&types.WorkflowExecution{
+					WorkflowID: "test-workflow-id",
+					RunID:      "test-run-id",
+				}).Times(1)
+				mockReplicationTask.EXPECT().getFirstEvent().Return(&types.HistoryEvent{
+					EventType: &workflowExecutionStartedType,
+				}).Times(1)
+			},
+			applyStartEventsFnAffordance: func() func(
```

Comment:
nit: you can just assign the function value returned by the func to the field, which makes the code easier to understand

Reply:
To nit: this is a mock you're assigning more than an affordance I think? it's a thing that you're expecting to get called as part of the test run?